### PR TITLE
handle minipage inside tabular

### PR DESCRIPTION
**Pull request by u-fischer**
fixes issue latex3/tagging-project#37

# Internal housekeeping

## Status of pull request

- Ready to merge

## Checklist of required changes before merge will be approved
- [x] Test file(s) added
- [x] Version and date string updated in changed source files
- [ ] Relevant `\changes` entries in source included
- [x ] Relevant `changes.txt` updated
- [ ] Rollback provided (if necessary)?
- [ ] `ltnewsX.tex` (and/or `latexchanges.tex`) updated

### CODE REVIEW

File: required/latex-lab/latex-lab-minipage.dtx [hunk 1]
```diff
@@ -126,6 +126,12 @@
       \hsize\@tempdima
       \textwidth\hsize \columnwidth\hsize
       \@parboxrestore
+%    \end{macrocode}
+% minipages are perhaps used in environments where the tagging of paragraphs
+% has been altered. So we try to restore this. TODO: this should
+% perhaps be done in \cs{@parboxrestore} directly. 
+%    \begin{macrocode}
+      \UseTaggingSocket{para/restore}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-minipage.dtx:134]: probably needs a % at the end of line

File: required/latex-lab/latex-lab-minipage.dtx [hunk 2]
```diff
@@ -165,7 +171,14 @@
...
   \setlength\@tempdima{#4}%
   \socket_use:n{tagsupport/parbox/before}
-  \@begin@tempboxa\vbox{\hsize\@tempdima\@parboxrestore#5\@@par}%
+  \@begin@tempboxa\vbox{\hsize\@tempdima\@parboxrestore
+%    \end{macrocode}
+% parbox are perhaps used in environments where the tagging of paragraphs
+% has been altered. So we try to restore this. TODO: this should
+% perhaps be done in \cs{@parboxrestore} directly. 
+%    \begin{macrocode}  
+  \UseTaggingSocket{para/restore}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-minipage.dtx:180]: same here:   %

File: required/latex-lab/latex-lab-minipage.dtx [hunk 3]
```diff
@@ -191,9 +204,11 @@
 \socket_new_plug:nnn{tagsupport/minipage/before}{tag/dflt}
   {
 %    \end{macrocode}
-% As there was a \cs{leavevmode} before that is a P/text open that we have to close.
+% As there was a \cs{leavevmode} before that is a P/text open that we have to close
+% if para tagging was active:
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-minipage.dtx:208]: is active

File: required/latex-lab/latex-lab-table.dtx [hunk 4]
```diff
@@ -546,20 +546,34 @@
 % \enquote{vmode} tables like longtable.
 %
 % \begin{plugdecl}{Table}
+%  Inside a TD or TR Part is not allowed as child. For structures that restore
+%  paragraph settings we therefore need a special plug that adjust the settings.
+%  Currently we set the para main tag to Div. This leads to double Div-structures
+%  but flattening the paragraph doesn't work, it errors if there is a list inside. 
+%    \begin{macrocode}
+\NewSocketPlug{tagsupport/para/restore}{Table}
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-table.dtx:554]: maybe this socket is better called "/adjust" if it is changing the para handling?

File: required/latex-lab/latex-lab-table.dtx [hunk 5]
```diff
@@ -546,20 +546,34 @@
...
+    \tl_set_eq:NN \l__tag_para_tag_tl\l__tag_para_tag_default_tl
+    \bool_set_true:N \l__tag_para_bool
+    \bool_set_false:N \l__tag_para_flattened_bool
+  }
+%    \end{macrocode}
+% \end{plugdecl}
+% 
+% \begin{plugdecl}{Table}
 %    Inside a table we currently only disable paratagging. We assume
 %    that these sockets are in an environment group, so there is no
```
Comment (FrankMittelbach) [inline required/latex-lab/latex-lab-table.dtx:566]: called in a group   

may not be an environment. I guess \parbox owuld have the same problem, wouldn't it?

**Review by FrankMittelbach — APPROVED** [review]
a few comments and questions

File: required/latex-lab/latex-lab-minipage.dtx (hunk 1 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-minipage.dtx:134]: it is an expl3 style, so % not needed (and there are \socket_use commands nearby). But looking at it I should change some of them to \UseTaggingSocket.

File: required/latex-lab/latex-lab-table.dtx (hunk 4 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-table.dtx:554]: well the general plan is to put this inside \@parboxrestore  at some time, so I think the name fits.

File: required/latex-lab/latex-lab-table.dtx (hunk 5 above)
Comment (u-fischer) [reply required/latex-lab/latex-lab-table.dtx:566]: well what was meant was that the table socket is called inside a tabular environment and so we don't have to reset at \end{tabular}. But I changed to "called in a group".

File: required/latex-lab/latex-lab-minipage.dtx (hunk 1 above)
Comment (FrankMittelbach) [reply required/latex-lab/latex-lab-minipage.dtx:134]: I only noticed the percents nearby so thought is is not within L3 layer code. Maybe in that case cleaning up the code near by should be done at some point (clearl not urgent)

File: required/latex-lab/latex-lab-table.dtx (hunk 4 above)
Comment (FrankMittelbach) [reply required/latex-lab/latex-lab-table.dtx:554]: well if it goes into parboxrestore then it isn't "table" specific in which case the plug name is wrong. If, however, we anticipate different plugs, adjusting  aspects of the paratagging then it still seems to to me that "adjust" is correct even if it is called in parbox restore

**Pull request merged**